### PR TITLE
ts/indicators-cci

### LINKED ISSUE #11842: Migrate `indicators/cci` to TypeScript

### PR DESCRIPTION
**Pull request by activist**
Closes #11842,  Migrated CciIndicator to TypeScript.

### CODE REVIEW

File: ts/indicators/cci.src.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    },
+    /**
+     * @lends Highcharts.Series#
+     */
+    {
+        getValues: function (
+            series: Highcharts.CciIndicator,
+            params: Highcharts.CciIndicatorParamsOptions
+        ): (boolean|Highcharts.IndicatorValuesObject) {
+            var period = (params.period as any),
```
Comment (bre1470) [inline ts/indicators/cci.src.ts]: Set explicit `number` type. Cast to `any` is not enough!

File: ts/indicators/cci.src.ts [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                xVal = (series.xData as any),
```
Comment (bre1470) [inline ts/indicators/cci.src.ts]: Same case here.

File: ts/indicators/cci.src.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+                CCI: Array<Array<number>> = [],
+                xData: Array<number> = [],
+                yData: Array<number> = [],
+                CCIPoint: number,
+                p: Array<number>,
+                len: number,
+                smaTP: number,
+                TPtemp: number,
+                meanDev: number,
+                i: (number|undefined);
```
Comment (bre1470) [inline ts/indicators/cci.src.ts]: Try without `undefined` and remove the `any` casts in the for-loop. For this the `period` variable has to be explicitly set to type `number`.

File: ts/indicators/cci.src.ts [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+                len = TP.push(TPtemp);
+                periodTP = TP.slice(len - period);
+
+                smaTP = sumArray(periodTP) / period;
+                meanDev = meanDeviation(periodTP, smaTP) / period;
+
+                CCIPoint = ((TPtemp - smaTP) / (0.015 * meanDev));
+
+                CCI.push([xVal[(i as any) - 1], CCIPoint]);
+                xData.push(xVal[(i as any) - 1]);
```
Comment (bre1470) [inline ts/indicators/cci.src.ts]: Remove the `any` casts for `i`.

File: ts/indicators/cci.src.ts [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+ */
+declare global {
+    namespace Highcharts {
+        class CciIndicator extends SmaIndicator {
+            public data: Array<CciIndicatorPoint>;
+            public getValues(
+                series: Series,
+                params: CciIndicatorParamsOptions
+            ): (boolean|IndicatorValuesObject);
+            public options: CciIndicatorOptions;
```
[the remark this answers is not in the document]
Comment (activist) [reply ts/indicators/cci.src.ts]: Ok, I think I misunderstood the reflection stuff to some extent. Wasn't that supposed to be automatically resolved?

[the remark this answers is not in the document]
Comment (bre1470) [reply ts/indicators/cci.src.ts]: No, this is about inheritence. If you do not override the property, it contains the type from the parent.

File: ts/indicators/cci.src.ts [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+import H from '../parts/Globals.js';
+
+/**
+ * Internal types
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        class CciIndicator extends SmaIndicator {
```
Comment (bre1470) [inline ts/indicators/cci.src.ts]: Needs to be renamed to `CCIIndicator` (and `SMAIndicator`).

Comment (bre1470) [reply ts/indicators/cci.src.ts]: Will be done in a separate PR.

**Pull request merged**